modules/tensor_mechanics/src/kernels/WeakPlaneStress.C

```diff
@@ -19,23 +21,50 @@ validParams<WeakPlaneStress>()
 {
   InputParameters params = validParams<Kernel>();
   params.addClassDescription("Plane stress kernel to provide out-of-plane strain contribution");
+  params.addCoupledVar("displacements",
+                       "The string of displacements suitable for the problem statement");
+  params.addCoupledVar("temperature",
+                       "The name of the temperature variable used in the "
+                       "ComputeThermalExpansionEigenstrain.  (Not required for "
+                       "simulations without temperature coupling.)");
+  params.addParam<std::string>(
+      "thermal_eigenstrain_name",
+      "thermal_eigenstrain",
+      "The eigenstrain_name used in the ComputeThermalExpansionEigenstrain.");
   params.addParam<std::string>("base_name", "Material property base name");
 
   MooseEnum direction("x y z", "z");
-  params.addParam<MooseEnum>("direction", direction, "The out of plane direction");
+  params.addParam<MooseEnum>(
+      "direction", direction, "The direction of the out-of-plane strain variable");
```

Comment:
I'd deprecate 'direction' in favor of 'out_of_plane_strain_direction' because it's more descriptive and consistent with what you added to StressDivergenceTensors.
```
params.addDeprecatedParam<MooseEnum>("direction", direction, "The direction of the out-of-plane strain variable", "Use the new parameter name 'out_of_plane_strain_direction'");
params.addParam<MooseEnum>("out_of_plane_strain_direction", direction, "The direction of the out-of-plane strain variable");
```
In the initialization list, you'll need something like this:
```
_direction(isParamSetByUser("direction") ? getParam<MooseEnum>("direction")) : getParam<MooseEnum>("out_of_plane_strain_direction")),
```
Finally, in the body of the constructor:
n the initialization list, you'll need something like this:
```
if (isParamSetByUser("direction") && isParamSetByUser("out_of_plane_strain_direction"))
  mooseError("Cannot specify both 'direction' and 'out_of_plane_strain_direction'"); 
```